launcher/ui/widgets/InfoFrame.cpp

```diff
@@ -1,54 +1,70 @@
-*  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-*  GNU General Public License for more details.
-*
-*  You should have received a copy of the GNU General Public License
-*  along with this program.  If not, see <https://www.gnu.org/licenses/>.
-*
-* This file incorporates work covered by the following copyright and
-* permission notice:
-*
-*      Copyright 2013-2021 MultiMC Contributors
-*
-*      Licensed under the Apache License, Version 2.0 (the "License");
-*      you may not use this file except in compliance with the License.
-*      You may obtain a copy of the License at
-*
-*          http://www.apache.org/licenses/LICENSE-2.0
-*
-*      Unless required by applicable law or agreed to in writing, software
-*      distributed under the License is distributed on an "AS IS" BASIS,
-*      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-*      See the License for the specific language governing permissions and
-*      limitations under the License.
-*/
-
+ *  Prism Launcher - Minecraft Launcher
+ *  Copyright (c) 2022 flowln <flowlnlnln@gmail.com>
+ *  Copyright (c) 2023 Trial97 <alexandru.tripon97@gmail.com>
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, version 3.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * This file incorporates work covered by the following copyright and
+ * permission notice:
+ *
+ *      Copyright 2013-2021 MultiMC Contributors
+ *
+ *      Licensed under the Apache License, Version 2.0 (the "License");
+ *      you may not use this file except in compliance with the License.
+ *      You may obtain a copy of the License at
+ *
+ *          http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *      Unless required by applicable law or agreed to in writing, software
+ *      distributed under the License is distributed on an "AS IS" BASIS,
+ *      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *      See the License for the specific language governing permissions and
+ *      limitations under the License.
+ */
+
+#include <QLabel>
 #include <QMessageBox>
+#include <QToolTip>
 
 #include "InfoFrame.h"
 #include "ui_InfoFrame.h"
 
 #include "ui/dialogs/CustomMessageBox.h"
 
-InfoFrame::InfoFrame(QWidget *parent) :
-    QFrame(parent),
-    ui(new Ui::InfoFrame)
+void setupLinkTooTip(QLabel* label)
+{
+    QObject::connect(label, &QLabel::linkHovered, [label](const QString& link) {
+        if (!link.isEmpty() && !link.startsWith("http"))
```

Comment:
Error-prone

Reply:
Well I need to check for a link starting `http` because there are places where the link is a custom string that instead of opening a web page does a QT action.
Can you suggest a better way to determine if the href is a web link?

Reply:
i'm curious, assuming the mod is downloadable from either Modrinth or Curseforge, is it possible to make use of the url used to download the mod itself to extrapolate the expected url for the mod page?

Reply:
ah it looks like you may already be doing that actually, apologies! not familiar with this codebase much atm

Reply:
> Well I need to check for a link starting `http` because there are places where the link is a custom string that instead of opening a web page does a QT action. Can you suggest a better way to determine if the href is a web link?

You probably want links starting with http:// or https://

Reply:
I did one better